generic3g/specs/FieldSpec.F90

```diff
@@ -245,36 +246,36 @@ subroutine set_field_default(rc)
          real(kind=ESMF_KIND_R8), pointer :: x_r8_1d(:),x_r8_2d(:,:),x_r8_3d(:,:,:),x_r8_4d(:,:,:,:)
          integer :: status, rank
 
-         call ESMF_FieldGet(this%payload,rank=rank,_RC) 
+         call ESMF_FieldGet(this%payload,rank=rank,_RC)
          if (this%typekind == ESMF_TYPEKIND_R4) then
             if (rank == 1) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r4_1d,_RC)
-               x_r4_1d = this%default_value   
+               x_r4_1d = this%default_value
             else if (rank == 2) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r4_2d,_RC)
-               x_r4_2d = this%default_value   
+               x_r4_2d = this%default_value
             else if (rank == 3) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r4_3d,_RC)
-               x_r4_3d = this%default_value   
+               x_r4_3d = this%default_value
             else if (rank == 4) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r4_4d,_RC)
-               x_r4_4d = this%default_value   
+               x_r4_4d = this%default_value
             else
                _FAIL('unsupported rank')
             end if
          else if (this%typekind == ESMF_TYPEKIND_R8) then
             if (rank == 1) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r8_1d,_RC)
-               x_r8_1d = this%default_value   
+               x_r8_1d = this%default_value
             else if (rank == 2) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r8_2d,_RC)
-               x_r8_2d = this%default_value   
+               x_r8_2d = this%default_value
             else if (rank == 3) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r8_3d,_RC)
-               x_r8_3d = this%default_value   
+               x_r8_3d = this%default_value
             else if (rank == 4) then
                call ESMF_FieldGet(this%payload,farrayptr=x_r8_4d,_RC)
-               x_r8_4d = this%default_value   
+               x_r8_4d = this%default_value
```

Comment:
Please open a ticket to have this somehow moved into "Darian's" rank/kind agnostic layer.  (If possible.)

Reply:
I'll change this in this PR just so trivial to change.

Reply:
Well tried changing to use the function that should do this, but made test fail so I'll futz with this later.